### ISSUE TITLE
New Rule: check for only one next in router guards

### ISSUE THREAD

**Issue by NeonWizard** [vuejs/eslint-plugin-vue#1457]
### Please describe what the rule should do:

From the wiki page:

> Make sure that the next function is called exactly once in any given pass through the navigation guard. It can appear more than once, but only if the logical paths have no overlap, otherwise the hook will never be resolved or produce errors.

The linter should make sure that for every `beforeEnter`, `beforeEach`, etc, `next()` can only be called once and is protected by if statements.

### What category should the rule belong to?

- [ ] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

### Provide 2-3 code examples that this rule should warn about:

```js
// BAD
router.beforeEach((to, from, next) => {
  if (to.name !== 'Login' && !isAuthenticated) next({ name: 'Login' })
  // if the user is not authenticated, `next` is called twice
  next()
})
```
```js
// GOOD
router.beforeEach((to, from, next) => {
  if (to.name !== 'Login' && !isAuthenticated) next({ name: 'Login' })
  else next()
})
```

**Additional context**
